sharding/handler/shard_tracker.py

```diff
@@ -84,69 +87,122 @@ def _get_logs_by_notary(self,
     # Basic functions to get emitted logs
     #
     @to_list
-    def get_register_notary_logs(self) -> Generator[LogParser, None, None]:
-        logs = self._get_logs_by_notary(event_name='RegisterNotary', notary=None)
+    def get_register_notary_logs(self,
+                                 *,
+                                 from_period: Union[int, str]=None,
+                                 to_period: Union[int, str]=None
+                                 ) -> Generator[LogParser, None, None]:
+        from_block = None if from_period is None else from_period * self.config['PERIOD_LENGTH']
+        to_block = None if to_period is None else (to_period + 1) * self.config['PERIOD_LENGTH'] - 1
```

Comment:
These two lines duplicate in multiple functions. I think it's better to refactor it.

Reply:
And we can make a clear function, I'd unvote for using ternary operator here.